test/e2e/mount_test.go

```diff
@@ -641,3 +650,88 @@ func TestMountRotateSecret(t *testing.T) {
 
 	// TODO: Add checks for regional secret
 }
+
+// Execute a test job that mounts a extract secret and checks that the value is correct.
+func TestMountExtractSecret(t *testing.T) {
+	secretData := []byte(`{"user":"admin", "password":"password@1234"}`)
+
+	// Create test secret
+	secretFile := filepath.Join(f.tempDir, "secretExtractValue")
+	check(os.WriteFile(secretFile, secretData, 0644))
+	check(execCmd(exec.Command(
+		"gcloud", "secrets", "create", f.testExtractSecretID,
+		"--replication-policy", "automatic",
+		"--data-file", secretFile,
+		"--project", f.testProjectID,
+	)))
+
+	// Create a regional test secret
+	check(execCmd(exec.Command("gcloud", "config", "set", "api_endpoint_overrides/secretmanager",
+		"https://secretmanager."+f.location+".rep.googleapis.com/")))
+
+	check(execCmd(exec.Command(
+		"gcloud", "secrets", "create", f.testExtractSecretID,
+		"--location", f.location,
+		"--data-file", secretFile,
+		"--project", f.testProjectID,
+	)))
+	check(execCmd(exec.Command("gcloud", "config", "unset", "api_endpoint_overrides/secretmanager")))
+
+	podFile := filepath.Join(f.tempDir, "test-extract-key.yaml")
+	if err := replaceTemplate("templates/test-extract-key.yaml.tmpl", podFile); err != nil {
+		t.Fatalf("Error replacing pod template: %v", err)
+	}
+
+	if err := execCmd(exec.Command("kubectl", "apply", "--kubeconfig", f.kubeconfigFile,
+		"--namespace", "default", "-f", podFile)); err != nil {
+		t.Fatalf("Error creating job: %v", err)
+	}
+
+	// As a workaround for https://github.com/kubernetes/kubernetes/issues/83242, we sleep to
+	// ensure that the job resources exists before attempting to wait for it.
+	time.Sleep(5 * time.Second)
```

Comment:
Did you face any errors ?

Reply:
We initially used time.Sleep(5) to ensure consistency with other tests, but after testing without it, we found that it works fine, so we’re removing it.